rust/tool/tests/common/mod.rs

```diff
@@ -205,3 +211,34 @@ pub fn verify_signature(path: &Path) -> Result<bool> {
 pub fn count_files(path: &Path) -> Result<usize> {
     Ok(fs::read_dir(path)?.count())
 }
+
+fn hash<'key, T: Serialize>(inputs: impl IntoIterator<Item = (&'key str, T)>) -> Result<String> {
```

Comment:
This looks like duplicated?

Reply:
The problem is that this implementation is quite deeply hidden in the install code, but then relied on by the test. The old `nixos_path` was duplicated as well. I can probably refactor this if desired.